### PR TITLE
chore: ban `Instant::now` via semgrep

### PR DESCRIPTION
**Pull request by dpc**

### CODE REVIEW

**Review by Maan2003 — APPROVED** [review]

File: .semgrep.all.yaml [hunk 1]
```diff
@@ -8,6 +8,22 @@ rules:
 
   severity: WARNING
 
+- id: ban-instant-now
+  languages:
+    - rust
+  message: Use fedimint_core::time::now instead of std::time::Instant::now for better wasm compat.
```
Comment (Maan2003) [inline .semgrep.all.yaml:14]: not correct replacement

File: .semgrep.all.yaml [hunk 2]
```diff
@@ -8,6 +8,22 @@ rules:
...
+  pattern: std::time::Instant::now
```
[the remark this answers is not in the document]
Comment (dpc) [reply .semgrep.all.yaml:15]: Ooooohhhh...

**Comment by Maan2003** [comment]
is this getting ignored somehow?
```
fm (97d1200) rg -w 'Instant' fedimint-core

fedimint-core/src/task.rs
9:use std::time::{Duration, Instant};
399:    pub async fn sleep_until(deadline: Instant) {
449:    pub async fn sleep_until(deadline: Instant) {
450:        sleep(deadline.saturating_duration_since(Instant::now())).await
```

**Comment by dpc** [comment]
> fedimint-core/src/task.rs

```
#[cfg(not(target_family = "wasm"))]
mod imp {
```

```
#[cfg(target_family = "wasm")]
mod imp {
```

Maybe due to this? BTW. How does it work? I thought `Instant` doesn't work in wasm ... so why is it in conditional compilation for ... wasm?

Oh well. We'll take what we can.

**Review by Maan2003 — APPROVED** [review]

**Pull request merged**